app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -1186,24 +1186,34 @@ private int levelToSeekbarProgress(double level) {
      * pokemon evolution and level set by the user.
      */
     private void populateAdvancedInformation(IVScanResult ivScanResult) {
+
         double goalLevel = seekbarProgressToLevel(expandedLevelSeekbar.getProgress());
-        int intSelectedPokemon =
+        int spinnerSelection =
                 extendedEvolutionSpinner.getSelectedItemPosition(); //which pokemon is selected in the spinner
         ArrayList<Pokemon> evolutionLine = pokeInfoCalculator.getEvolutionLine(ivScanResult.pokemon);
+        extendedEvolutionSpinnerAdapter.updatePokemonList(evolutionLine);
+
+        Pokemon selectedPokemon = ivScanResult.pokemon;
 
-        Pokemon selectedPokemon;
-        if (intSelectedPokemon == -1) {
-            selectedPokemon = ivScanResult.pokemon;//if initialising list, act as if scanned pokemon is marked
-            for (int i = 0; i < evolutionLine.size(); i++) {
-                if (evolutionLine.get(i).number == selectedPokemon.number) {
-                    intSelectedPokemon = i;
-                    extendedEvolutionSpinner.setSelection(intSelectedPokemon);
+        if (spinnerSelection == -1) { // initializing the estimation box
+            boolean hasEvolution = selectedPokemon.evolutions.size() > 0;
+            int selectIndex = 0;
+            for (Pokemon evoPoke : evolutionLine) { //find the index of the pokemon in evolution line
+                if (evoPoke.number == selectedPokemon.number) {
                     break;
                 }
+                selectIndex++;
             }
+            if (hasEvolution) { //select the next pokemon in the spinner if evolution exists
+                selectIndex++;
+            }
+            extendedEvolutionSpinner.setSelection(selectIndex);
+
         } else {
-            if (evolutionLine.size() > intSelectedPokemon) {
-                selectedPokemon = evolutionLine.get(intSelectedPokemon);
+            //spinner has been previously initialized, dont change selection unless user managed to bug around
+            // interface and tries to set the selection outside the possible evolution line.
```

Comment:
I'd avoid having "bug around interface"—somebody like me seeing the comment would consider removing the check. But IIRC this happens easily, that is whenever you scan some pokemon A and then another pokemon B, if B's evolution line is smaller than A's.